packages/next/src/lib/metadata/resolvers/resolve-opengraph.ts

```diff
@@ -83,7 +83,8 @@ function resolveAndValidateImage(
     // devtools.
     const shouldWarn =
       !isUsingVercelSystemEnvironmentVariables &&
-      (process.env.NODE_ENV === 'production' || !isStaticMetadataRouteFile)
+      ((process.env.NODE_ENV === 'production' && !metadataBase) ||
```

Comment:
shouldn't the `!metadataBase` check be on its own line, since the warning is about not setting a `metadataBase`? ie: 

```
    const shouldWarn =
      !isUsingVercelSystemEnvironmentVariables &&
      !metadataBase &&
      (process.env.NODE_ENV === 'production' || !isStaticMetadataRouteFile)
```